langchain/document_loaders/directory.py

```diff
@@ -75,7 +75,34 @@ def load(self) -> List[Document]:
         """Load documents."""
         p = Path(self.path)
         docs: List[Document] = []
-        items = list(p.rglob(self.glob) if self.recursive else p.glob(self.glob))
+
+        # Glob specified file pattern(s)
+        if self.file_pattern:
+            # Add "." prefix in case user did not include it
+            self.file_pattern = {
```

Comment:
Technically can use a List too. The performance difference between sets and lists is generally negligible for small sizes. Probably only in the thousands would you then see sets being faster. And in our case, I don't imagine most use cases having thousands of file patterns.

But anyhoos, no reason not to use sets as well, no?